### PR TITLE
Individual part uploads shouldn't have the server-side encryption hea…

### PR DESCRIPTION
**Pull request by absurdfarce**
…der supplied.

### CODE REVIEW

**Comment by absurdfarce** [comment]
@arr-ee This is a _provisional_ pull request for the jclouds side of OPSC-7247.  It was created at Matt's request to demonstrate what such a solution might look like.  It is _completely_ untested so please _don't_ merge until further testing has been completed on that ticket!

File: providers/aws-s3/src/main/java/org/jclouds/aws/s3/blobstore/internal/AWSS3PutOptionsToPutObjectOptions.java [hunk 1]
```diff
@@ -27,16 +27,45 @@
...
+		}
+
+		/* Part options are always fixed at defaults due to OPSC-7247.  This may change if we start supporting
+		 * additional S3 options (ACLs etc.) in future ops. */
+		public PutObjectOptions getPartOptions() {
+			return PutObjectOptions.DEFAULTS;
+		}
+	}
+
+	/* We can get away with this because yes/no to server-side encryption is the only axis of variation at this point. */
```
Comment (nickmbailey) [inline providers/aws-s3/src/main/java/org/jclouds/aws/s3/blobstore/internal/AWSS3PutOptionsToPutObjectOptions.java:58]: as in this is the only header that we expect to attach to the requests or this is the only header that can be attached at this point in the code?

File: apis/s3/src/main/java/org/jclouds/s3/options/PutObjectOptions.java [hunk 2]
```diff
@@ -104,16 +104,16 @@ public void setHeaderTag(@Named(PROPERTY_HEADER_TAG) String headerTag) {
 
 		protected Builder() {
 
-			this.useServerSideEncryption = DEFAULT_SERVER_SIDE_ENCRYPTION;
-			this.acl = DEFAULT_ACCESS_POLICY;
+			this.useServerSideEncryption = DEFAULTS.usesServerSideEncryption();
+			this.acl = DEFAULTS.getAcl();
```
Comment (nickmbailey) [inline apis/s3/src/main/java/org/jclouds/s3/options/PutObjectOptions.java]: this change seems unrelated to the rest..,

File: providers/aws-s3/src/main/java/org/jclouds/aws/s3/blobstore/internal/AWSS3PutOptionsToPutObjectOptions.java (hunk 1 above)
Comment (absurdfarce) [inline providers/aws-s3/src/main/java/org/jclouds/aws/s3/blobstore/internal/AWSS3PutOptionsToPutObjectOptions.java:58]: More like the second.  When I implemented the logic for generating main and part options it became clear that the only way they varied at the moment was through the server-side encryption header; for everything else we were just using whatever was defined in the input PutOptions object.  This comment was meant to highlight that point.

In the future we _may_ find ourselves adding new headers (or dealing with existing headers) which will also need modification when they're used in a main vs. part PutObjectOptions (or vice versa).  But for now the server-side encryption header is the only header which requires such customization, thus it's the only axis of variation.

File: apis/s3/src/main/java/org/jclouds/s3/options/PutObjectOptions.java (hunk 2 above)
Comment (absurdfarce) [inline apis/s3/src/main/java/org/jclouds/s3/options/PutObjectOptions.java]: Yeah, this was something of a general code cleanup.  I was just trying to make the builder re-use whatever my static DEFAULTS instance had for these values rather than assign them explicitly.

Honestly, I'd prefer not to have this DEFAULTS instance at all; far better to hide that away in the builder constructor and let everybody use the builder to get whatever is appropriate.  The original code was doing something like what's done here, but they were just less direct about it (and called their default instance NONE, the logic of which still escapes me).  I consolidated everything into the DEFAULTS instance to clarify the logic... but that was always something of an incremental step.
